src/routes/IndexRoute.vue

```diff
@@ -1,26 +1,15 @@
 <script lang="ts" setup>
 import DefaultLayout from '../layouts/DefaultLayout.vue';
-import { useI18n } from 'vue-i18n'
+import TrickOverview from '../components/TrickOverview.vue';
+import { ref } from 'vue';
 
-import messages from "../i18n/index"
-
-const i18n = useI18n({
-  messages,
-  useScope: "local"
-})  
-
-const {t} = i18n;
+const searchText = ref('')
 
 </script>
 
 
 <template>
-  <DefaultLayout :flex-spacer="false">
-    <div class="flex flex-1 flex-col justify-center  self-stretch ">
-      <div class="flex flex-row justify-center">
-        <h1 class="text-5xl font-bold dark:text-white transition-all duration-200 hover:-translate-y-3">{{ t('greeting') }}</h1>
-      </div>
-    </div>
+  <DefaultLayout :flex-spacer="true" @update-search-query="(searchQuery) => searchText = searchQuery">
+    <TrickOverview :searchText="searchText"/>
```

Comment:
This is the correct way to pass events

Reply:
Ok, so that means all the variables that are passed to some child components will appear in the `IndexRoute` and then passed as props to whereever they are needed. Aren't we gonna end up with lots of variables in here then?